test/end-to-end/test/test_editRecipe.js

```diff
@@ -0,0 +1,143 @@
+const Nightmare = require('nightmare');
+const expect = require('chai').expect;
+const EditRecipePage = require('../pages/editRecipe');
+const CreateComposPage = require('../pages/createCompos');
+const ToastNotifPage = require('../pages/toastNotif');
+const apiCall = require('../utils/apiCall');
+const pageConfig = require('../config');
+
+describe('Edit Recipe Page', function () {
+  this.timeout(15000);
```

Comment:
From [Mocha doc](https://mochajs.org/#timeouts): This is a suite-level timeout, which may be applied to entire test “suites”. This will be inherited by all nested suites and test-cases that do not override the value.

Reasons to put timeout setting there:

1. The position of all timeout setting is at the top level suite, so that it can be inherited by all nested suites and test-cases.

2. And every suite should have its own timeout according to its case. So I put a common number, say 15000ms, here.

3. Explicit is a better choice for settings. It is easy for case writer to adjust based on their requirement. Just as I put the case expectation at the top line of each case block.